src/session/session_api.c

```diff
@@ -1583,55 +1584,103 @@ __wt_session_range_truncate(
     trunc_info->stop = stop;
     trunc_info->orig_start_key = orig_start_key;
     trunc_info->orig_stop_key = orig_stop_key;
-    if (uri != NULL)
-        trunc_info->uri = uri;
-    else if (start != NULL)
-        trunc_info->uri = start->internal_uri;
-    else {
-        WT_ASSERT(session, stop != NULL);
-        trunc_info->uri = stop->internal_uri;
-    }
+    trunc_info->uri = actual_uri;
+
+    /*
+     * We would like to use bounded cursors if possible, so check whether they are supported. We
```

Comment:
This comment is still quite verbose, it could be:
```
Use bounded cursors if possible, check if they are supported. Fixed length column store doesn't support bounded cursors, additionally skip complex types such as column groups and index. Those tables may or may not s
```